internal/controller/nats/integrationtests/validation/integration_test.go

```diff
@@ -93,3 +99,70 @@ func Test_Validate_CreateNatsCR(t *testing.T) {
 		})
 	}
 }
+
+func Test_NATSCR_Defaulting(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name string
+		// We use Unstructured instead of NATS to ensure that all undefined properties are nil and not Go defaults.
+		givenUnstructuredNATS unstructured.Unstructured
+		wantMatches           gomegatypes.GomegaMatcher
+	}{
+		{
+			name: "defaulting",
+			givenUnstructuredNATS: unstructured.Unstructured{
+				Object: map[string]interface{}{
+					"kind":       "NATS",
+					"apiVersion": "operator.kyma-project.io/v1alpha1",
+					"metadata": map[string]interface{}{
+						"name":      "name-defaulting01",
+						"namespace": "namespace-defaulting01",
+					},
+				},
+			},
+			wantMatches: gomega.And(
+				natsmatchers.HaveSpecClusterSize(3),
+				natsmatchers.HaveSpecResources(corev1.ResourceRequirements{
+					Limits: corev1.ResourceList{
+						"cpu":    resource.MustParse("20m"),
+						"memory": resource.MustParse("64Mi"),
+					},
+					Requests: corev1.ResourceList{
+						"cpu":    resource.MustParse("5m"),
+						"memory": resource.MustParse("16Mi"),
+					},
+				}),
+				natsmatchers.HaveSpecLoggingTrace(false),
+				natsmatchers.HaveSpecLoggingDebug(false),
+				natsmatchers.HaveSpecJetsStreamMemStorage(v1alpha1.MemStorage{
+					Enabled: false,
+					Size:    resource.MustParse("20Mi"),
+				}),
+				natsmatchers.HaveSpecJetStreamFileStorage(v1alpha1.FileStorage{
+					StorageClassName: "default",
+					Size:             resource.MustParse("1Gi"),
+				}),
+			),
+		},
```

Comment:
As per your comment:
```
Default values are defined at multiple levels to ensure that values are set independently of what level of NATS is
defined or gets deleted. For example, spec.cluster.size will get set to a default value whether spec.cluster.size,
spec.cluster or spec gets deleted.
```

Can we add more test cases to cover the multiple levels?

Reply:
valid! will do!